config/routes.rb

```diff
@@ -19,8 +19,10 @@
     root to: "theses#index"
   end
 
+  get 'thesis/submit', to: 'thesis#submit', as: 'submit'
```

Comment:
`submit` feels more like the action of when you click the form rather than the `new or edit` logic it really is.

I wonder if `start` would be clearer? It's the starting point for either path. Not sure. Non-blocking but please do consider if `start` or something else might be more clear when we look back at this work during future maintenance.

Reply:
Yeah, this seems like a better name - will change.